### PR TITLE
Task/add test coverage/606

### PR DESCRIPTION
**Pull request by anthonyhashemi**
https://trello.com/c/Z8QT4CCd/606-add-test-coverage-to-enrichment-repo

This PR:

- adds `coverage.py` to the repo and updates CI and readme to use `coverage.py`
- deletes `tests/runner.py` and simplifies tests and updates docs to recommend to run tests with `python -m unittest`
- deletes old redundant tests
- fix and simplify old, previously nonrunning test suite

### CODE REVIEW

File: tests/test_pipeline_components/test_replacer.py
Comment (anthonyhashemi) [inline tests/test_pipeline_components/test_replacer.py:1]: This file was not being run in CI due to not being included in the runner.py file.

It was old and needed updating.

Realised that we are just testing the `write_replacements_file` and so having 1 test for each type of replacement in the repo was unnecessary. Instead we now just have one generic test which shows that this currently actually allows us to have various namedtuples in the same replacements list. Could be worth refactoring the function/code that calls it but not worth it for now.

File: docs/caselaw/case-law-annotator.md [hunk 1]
```diff
@@ -154,7 +154,7 @@ Crucially, the Case Law Annotator does not alter the original citation used by t
...
 |canonicalForm|Required|The template for the detected citation's canonical form, e.g. `[dddd] d1 WLR d2`. <br><br> The `canonicalForm` is used in combination with the `citationType` to resolve malformed citations to their canonical form, see [correction_strategies.py](/caselaw_extraction/correction_strategies.py).|
 |canonicalExample|Required|An example of a well-formed citation matched by the relevant family of rules. The main purpose of the `canonicalExample` is to provide an example of a well-formed citation for a given rule.|
-|matchExample|Required|The `matchExample` is an example of a citation intended to be detected by the corresponding rule. The main purpose of the `matchExample` is to test that a given rule does in fact match the intended citation style. The `matchExample` is used in automated tests that run whenever a [modified version of the Rules Manifest is uploaded to S3](/docs/caselaw/adding-new-citation-rules.md).|
```
Comment (anthonyhashemi) [inline docs/caselaw/case-law-annotator.md:157]: matchExample is actually match_example nowadays

File: tests/test_manifest.py
Comment (anthonyhashemi) [inline tests/test_manifest.py:1]: Deleted as its old and not passing and wasnt being run. An updated version of this test (`test_manifest`) is actually run in the `update_rules_processor` lambda function.

**Review by dragon-dxw — APPROVED** [review]

**Comment by dragon-dxw** [comment]
❤️

**Comment by anthonyhashemi** [comment]
Will add code cov to CI in separate PR

**Pull request merged**